src/main/groovy/org/boozallen/plugins/jte/init/PipelineDecorator.groovy

```diff
@@ -189,4 +191,25 @@ class PipelineDecorator extends InvisibleAction {
         return flowOwner.run().getParent()
     }
 
+    @SuppressWarnings("PropertyName")
+    static class JteBlockWrapper{
+        /**
+         * property name should match the config keys which are snake_case/underscore-separated
+         */
+        String pipeline_template = null
+        Boolean allow_scm_jenkinsfile = true
+        JteBlockWrapper(final LinkedHashMap config){
+            List<String> fields = this.getMetaClass().properties*.getName()
+            fields.each{ field ->
+                if(config.containsKey(field)){
+                    this.setProperty(field, config[field])
```

Comment:
Should we wrap this in a try catch? 

for example, with the following pipeline configuration: 
```groovy
jte{
  pipeline_template = true
}
```

this line will likely throw a ClassCast exception bc boolean can't be converted to the String type? 

so wrapping seetProperty in try/catch to have some nicer error logging of like "jte.pipline_template must be a String" or something?

Reply:
going to add the validation.

Reply:
added validation